src/tools/cephfs/cephfs-shell

```diff
@@ -773,6 +773,148 @@ class CephFSShell(Cmd):
+                    dest = args.dest_path.decode('utf-8').rstrip('/')
+                    dest_first_half = dest.encode('utf-8') + b'/'
+                    file = re.findall(r"[^/]+(?=/$|$)",
+                                      args.src_path.decode('utf-8'))
+                    # it will always match the last part of path, i.e
+                    # either a file or dir, length of list will never be
+                    # more than one. If so then it is violating something
+                    # and should not be allowed to proceed.
+                    if len(file) == 1:
+                        dest_second_half = file[0].encode('utf-8')
+                        dest_path = dest_first_half + dest_second_half
+                    else:
+                        set_exit_code_msg(msg="failed to create destination"
+                                              " path as more than one"
+                                              " file/dir matched with regex")
+                        return
+                # if it is a dir then take the dir and concatenate it
+                # with the destination path
+                # NOTE: If the dir in both source and destionation are same
+                # then it means dir link should not be inserted into the dir
+                # but the provided destination path should be followed
+                elif is_dir_exists(args.src_path):
+                    dir1 = re.findall(r"[^/]+(?=/$|$)",
+                                      args.src_path.decode('utf-8'))
+                    dir2 = re.findall(r"[^/]+(?=/$|$)",
+                                      args.dest_path.decode('utf-8'))
+                    if dir1[0] == dir2[0]:
+                        dest_path = args.dest_path
+                    else:
+                        dest = args.dest_path.decode('utf-8').\
+                            rstrip('/')
+                        dest_first_half = dest.encode('utf-8') + b'/'
+                        dest_second_half = dir1[0].encode('utf-8')
+                        dest_path = dest_first_half + dest_second_half
+
+            else:
+                # if the destination is not a file or a dir then:
+                # accept it as file so the end part of path cannot have
+                #    a `/` succeeding it.
+                test_path = args.dest_path.decode('utf-8')
+                if test_path[len(test_path) - 1] == '/':
+                    set_exit_code_msg(errno.ENOENT, f"'{test_path}': "
+                                                    f"No such file or "
+                                                    f"directory")
+                    return
+                else:
+                    dest_path = test_path
+        # sym link creating
+        if args.symbolic:
+            if args.force:
+                try:
+                    cephfs.unlink(dest_path)
+                except libcephfs.Error:
+                    pass
```

Comment:
Easier and cleaner approach would be to not to (generate and) catch libcephfs.Error in case where no file is present at destination_path but prevent such a case -
```suggestion
if args.force
    if stat(dest) == 0:
        cephfs.unlink()
    else: # dest path doesn't exists, yay!
        pass
```